src/PurchaseFactory.java

```diff
@@ -0,0 +1,31 @@
+import java.util.Scanner;
+
+public class PurchaseFactory {
+    private enum PurchaseKind {
+        GENERAL_PURCHASE {
+            Purchase getPurchase(Scanner sc) {
+                return new Purchase(sc);
+            }
+        },
+        PRICE_DIS_PURCHASE {
+            PriceDiscountPurchase getPurchase(Scanner sc) {
+                return new PriceDiscountPurchase(sc);
+            }
+        },
+        PERCENT_DIS_PURCHASE {
+            PercentDiscountPurchase getPurchase(Scanner sc) {
+                return new PercentDiscountPurchase(sc);
+            }
+        };
+        abstract Purchase getPurchase(Scanner sc);
```

Comment:
Is it correct to use abstract class here due to you can't create an object of abstract class?